src/response.rs

```diff
@@ -82,6 +82,22 @@ pub struct Response {
     compression: Option<Compression>,
 }
 
+enum Reader {
+    Buffered(Box<Cursor<Vec<u8>>>),
```

Comment:
~What does the `Box` do here? Isn't `Cursor<Vec<u8>>` enough since `Vec` will be on the heap anyway?~

Strike that. It's because we don't want to change `into_reader` signature further down.

Did you try going the route of having `Box<Reader>` in the struct and remove the internal boxes in the enum? Just curious if it gets less boxy.

Reply:
I didn't try that because of the variant `Connection(Box<dyn Read + Send + Sync + 'static>),`. This one needs to be boxed anyway (because I cannot use `dyn` on its own) and then the other variant ended with a `Box` for the reason you already found.

If we wanted to use concrete types in the enum, there would have to be more types because currently `Connection` is used for an open TCP socket as well as a compression wrapper on both `Cursor` and connection.